### DISCUSSION TITLE
Github Classroom Tests not running / no feedback PR created.

### DISCUSSION THREAD

**Discussion by srush** [community/community#136965] [GitHub Education]
I am setting up my github classroom for the semester. I have used it for the past 4 years and have set up the repo exactly like I normally do. However when I create the tests there is no longer a feedback PR for the students, and no tests are run. When I clear "Run Tests" I get a pop-up asking if I want to run the tests, but they do not run. 

Panicking a bit. Seems much more broken this year than in the past.

**Comment by GabrielAlmeidag** [comment]
@srush 😺👋👋👋

It seems the issue might be related to incorrect configuration or a problem with GitHub Classroom. First, ensure that all repositories and test configurations are set up correctly according to the current GitHub Classroom guidelines. Verify that the tests are properly configured and associated with the students' assignments or submissions. Check if there are any recent updates or changes to the platform that could affect test execution. If the configuration is correct and the problem persists, try unlinking and relinking the repositories or creating a new test environment to see if that resolves the issue. Additionally, review the GitHub Classroom documentation for any updates or changes.

**Reply by srush** [reply]
Can you give me links? Has anything changed since last year that would cause the same tests to stop working?

**Reply by GabrielAlmeidag** [reply]
Yes, there might be changes affecting your tests. To check, refer to these links:

[GitHub Classroom Documentation](https://docs.github.com/en/education)
[GitHub Classroom Changelog](https://docs.github.com/en/education/get-started-with-github-classroom/about-github-classroom)
[GitHub Actions Documentation](https://docs.github.com/en/actions)

These resources can help you determine if there have been any changes since last year that could be causing your tests to stop working.

**Reply by srush** [reply]
Yeah, it seems like it's just broken.

**Comment by chrisokasaki** [comment]
@srush Yes, things have changed as of the middle of last week, right in the middle of my first homework! I know there were warnings prior to that (although I can't find them at the moment).  One big thing I think was that the way they were automatically doing the feedback pull requests was becoming harder and harder to maintain. Another thing I think was that, going forward, the repos created via GitHub classroom would have the full history.  I've been playing with it a bit and recreating the feedback pull request looks possible but painful.  If I can't get that working soon, I'll probably just abandon GitHub classroom because the feedback pull requests were really the only thing keeping me here.

**Reply by srush** [reply]
I'm not sure I understand. They just removed this feature entirely? Why is there still a checkbox on assignments that includes this. 

Do you have recommendations for other similar services?

**Reply by chrisokasaki** [reply]
I don't really have any useful information.  I was mostly just validating that what you were experiencing was real.

**Comment by pinventado** [comment]
I'm reused an assignment from last semester, and I get inconsistent results. Mostly, the unit tests are not added to the student repository (.github folder is missing). I've had cases where the repository was created bu it was empty. Anyone having a similar issue?

**Reply by pinventado** [reply]
I ended up copying the .github folder from the Github's autograder from last semester and adding that as part of the starter code. Temporary fix, but I hope it gets worked out.

**Reply by srush** [reply]
This worked for me as well thanks.

**Comment by jdelfino** [comment]
There are numerous reports of automated tests and feedback PRs being totally broken right now. See:

https://github.com/orgs/community/discussions/136461
https://github.com/github/gh-classroom/issues/78
https://github.com/orgs/community/discussions/132094
etc...

I haven't seen any responses indicating the issue is being worked on, or even acknowledged, but some of the other threads do contain the workaround you noted here for getting tests running (creating your own .github folder).
